libraries/AP_Math/control.h

```diff
@@ -110,6 +110,9 @@ void shape_pos_vel_accel_xy(const Vector2p& pos_input, const Vector2f& vel_input
                             float vel_max, float accel_max,
                             float jerk_max, float dt, bool limit_total_accel);
 
+// limit_accel_xy limits the acceleration to prioritise acceleration perpendicular to the provided velocity vector.
+bool limit_accel_xy(const Vector2f& vel, Vector2f& accel, float accel_max);
```

Comment:
I think the function itself can be made "const" because no internal state changes.. maybe?